js/pulse.js

```diff
@@ -904,6 +907,10 @@
                 }
             };
 
+            var _onDeviceIdSet = function(event, deviceId) {
+                this._persistentId = deviceId
```

Comment:
If _onDeviceIdSet is called after loadMetadata (I'm not sure if that's possible now) this would overwrite the _persistentId.  And I think the pulse_persistent_id is supposed to win if you have both specified.